### PR TITLE
[AIRFLOW-2955] Fix kubernetes pod operator to set requests and limits on task pods

### PR DESCRIPTION
**Pull request by suensummit**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-2955
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Fix `kubernetes_pod_operator` to set resources requests/limits with a dictionary instead of passing an entire class `Resource()` to the operator. Also add test method `test_pod_resources` for it.

### Tests

- [x] My PR adds the following unit tests: `test_pod_resources`

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by feng-tao** [comment]
PTAL @dimberman

**Comment by suensummit** [comment]
Thanks @feng-tao 
Is there anything need to fix yet? @dimberman

**Comment by suensummit** [comment]
Just resolved conflicts that due to merge order; PTAL @dimberman thanks.

File: airflow/contrib/operators/kubernetes_pod_operator.py [hunk 1]
```diff
@@ -140,6 +142,15 @@ def execute(self, context):
         except AirflowException as ex:
             raise AirflowException('Pod Launching failed: {error}'.format(error=ex))
 
+    def _set_resources(self, resources):
+        inputResource = Resources()
+        if resources is not None:
+            for item in resources.keys():
+                setattr(inputResource, item, resources[item])
+        else:
+            pass
```
Comment (galuszkak) [inline airflow/contrib/operators/kubernetes_pod_operator.py]: That `else` is unnecessary?

**Review by galuszkak — CHANGES_REQUESTED** [review]

**Comment by dimberman** [comment]
Besides the nit from @galuszkak this all LGTM. Please address that comment, rebase/clean up commits, and then I'd be glad to merge :).

**Comment by suensummit** [comment]
@dimberman Just rebase and clean up commits, please check it after the CI passed, thanks.

File: airflow/contrib/operators/kubernetes_pod_operator.py [hunk 2]
```diff
@@ -140,6 +142,15 @@ def execute(self, context):
         except AirflowException as ex:
             raise AirflowException('Pod Launching failed: {error}'.format(error=ex))
 
+    def _set_resources(self, resources):
+        inputResource = Resources()
+        if resources is not None:
```
Comment (dimberman) [inline airflow/contrib/operators/kubernetes_pod_operator.py]: you can just write "if resources:" here

**Review by dimberman — CHANGES_REQUESTED** [review]
Just mine and @galuszkak 's nits and this is good to go

File: airflow/contrib/operators/kubernetes_pod_operator.py (hunk 2 above)
Comment (suensummit) [reply airflow/contrib/operators/kubernetes_pod_operator.py]: OK cool, thanks. @dimberman

Comment (suensummit) [reply airflow/contrib/operators/kubernetes_pod_operator.py]: Hi @dimberman PTAL thanks

**Review by dimberman — APPROVED** [review]

**Pull request merged**